File: extract_routes.py
#!/usr/bin/env python3
import re

# Read the route references file
with open('route_references.txt', 'r', encoding='utf-8', errors='ignore') as f:
    content = f.read()

# Find all route names - match route('name') or route("name")
pattern = r"route\(['\"]([^'\"]+)['\"]"
matches = re.findall(pattern, content)
route_names = sorted(set(matches))

# Write to file
with open('view_route_names.txt', 'w') as f:
    for name in route_names:
        f.write(name + '\n')

print('\n'.join(route_names[:100]))
print(f'\n... total {len(route_names)} routes')

File: extract_routes.php
<?php
$content = file_get_contents('route_references.txt');

// Convert from UTF-16LE to UTF-8
$content = mb_convert_encoding($content, 'UTF-8', 'UTF-16LE');

// Remove BOM if present
$content = preg_replace('/^\xEF\xBB\xBF/', '', $content);

// Find all route names - match route('name') or route("name")
preg_match_all("/route\(\s*['\"]([^'\"]+)['\"]/", $content, $matches);

$route_names = array_unique($matches[1]);
sort($route_names);

// Write to file
file_put_contents('view_route_names.txt', implode("\n", $route_names));

echo "Found " . count($route_names) . " unique routes in views" . PHP_EOL;
echo "First 50:" . PHP_EOL;
echo implode("\n", array_slice($route_names, 0, 50));
?>
